sound/soc/sof/intel/hda-pcm.c

```diff
@@ -32,6 +32,10 @@ static bool hda_always_enable_dmi_l1 = IS_ENABLED(CONFIG_SND_SOC_SOF_HDA_ALWAYS_
 module_param_named(always_enable_dmi_l1, hda_always_enable_dmi_l1, bool, 0444);
 MODULE_PARM_DESC(always_enable_dmi_l1, "SOF HDA always enable DMI l1");
 
+static bool hda_disable_rewinds = IS_ENABLED(CONFIG_SND_SOC_SOF_HDA_DISABLE_REWINDS);
```

Comment:
is there any specific reason to have this parameter (and another one in an earlier patch) both in Kconfig and as a module parameter? Wouldn't just a module parameter be enough?

Reply:
Ok, but apart from being a common approach, who builds custom kernels these days? And distros will have this parameter set one way or another for all their kernels anyway. Much easier to change under modprobe.d

Reply:
but in fact, in this specific case, you write, that it (also) depends on the audio user-space (PA vs. PipeWire vs...) and it's exactly distros who will configure that default user-space. But if the user opts for the non-default option? E.g. if the distro's default is PipeWire so they enable this feature. But then if the user wants to install and use PulseAudio instead, they'll have to manually add the kernel parameter? Or the distro will have to know to do that during PA installation... Cannot this be done automatically?

Reply:
It's not realistic to assume that kernel configurations will change if the user decides to revert to PulseAudio. Either the user will have to edit /etc/modprobe.d/, or the script to switch from PipeWire to PulseAudio does it. see https://fedoraproject.org/wiki/Changes/DefaultPipeWire

Or they don't use the parameter to force rewinds to be enabled. In practice I don't see the difference.